src/test/java/org/sagebionetworks/bridge/models/files/FileReferenceTest.java

```diff
@@ -0,0 +1,57 @@
+package org.sagebionetworks.bridge.models.files;
+
+import static org.sagebionetworks.bridge.TestConstants.GUID;
+import static org.sagebionetworks.bridge.TestConstants.TIMESTAMP;
+import static org.sagebionetworks.bridge.config.Environment.PROD;
+import static org.sagebionetworks.bridge.config.Environment.UAT;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.config.Environment;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class FileReferenceTest {
+    
+    private static final String HREF = BridgeConfigFactory.getConfig()
+            .getHostnameWithPostfix("docs") + "/" + GUID + "." + TIMESTAMP.getMillis();
+
+    @Test
+    public void equalsVerifier() {
+        EqualsVerifier.forClass(FileReference.class).allFieldsShouldBeUsed().verify();
+    }
+    
+    @Test
+    public void nullValues() throws Exception {
+        FileReference ref = BridgeObjectMapper.get().readValue("{}", FileReference.class);
+        assertNull(ref.getGuid());
+        assertNull(ref.getCreatedOn());
+        assertNull(ref.getHref());
+    }
```

Comment:
I think we need two separate tests for null guid and null createdOn, because they're two separate branches in getHref().

Also, is it necessary to use JSON deserialization for this test? It would be more clear to call the constructor directly.

Reply:
Split and removed Jackson here.